specs/agents/tracing-spans-service-target.md

```diff
@@ -0,0 +1,208 @@
+
+However, it does not replace [destination fields](https://www.elastic.co/guide/en/ecs/current/ecs-destination.html)
+which are used for APM/SIEM integration and focus on network-level attributes.
+
+- [span.context.destination.address](https://www.elastic.co/guide/en/ecs/current/ecs-destination.html#field-destination-address)
+- [span.context.destination.port](https://www.elastic.co/guide/en/ecs/current/ecs-destination.html#field-destination-port)
+
+APM Agents should now use fields described below in `span.context.service.target`, those are a subset
+of [ECS Service fields](https://www.elastic.co/guide/en/ecs/current/ecs-service.html).
+
+- `span.context.service.target.type` : [ECS service.type](https://www.elastic.co/guide/en/ecs/current/ecs-service.html#field-service-type)
+  , optional.
+- `span.context.service.target.name` : [ECS service name](https://www.elastic.co/guide/en/ecs/current/ecs-service.html#field-service-name)
+  , optional, ignored if `span.context.service.target.type` is not provided.
+
+Alignment to ECS provides the following benefits:
+
+- Easier correlation with other data sources like logs and metrics if they also rely on those service fields.
+- Posible future extension with other
+  ECS [service fields](https://www.elastic.co/guide/en/ecs/current/ecs-service.html) to provide higher-granularity
+- Bring APM Agents intake and data stored in ES closer to ECS
+
+On agents side, it splits the values that were previously written into `span.destination.service.resource` in distinct
+fields.It provides a generic way to provide higher-granularity for service map and service dependencies.
+
+#### APM Agents
+
+There are a few features in APM Agents relying on `span.destination.service.resource`:
+
+- Dropped spans metrics
+- Compressed spans
+- Agent API might provide a way to manually set the `span.destination.service.resource`, if such API exists we should provide an equivalent to enable setting `span.service.target.type` and `span.service.target.name`.
+
+#### APM Server
+
+The following features rely on `span.destination.service.resource` field:
+
+- APM Map
+- APM Services Dependencies
+- Dropped spans metrics (indirect dependency)
+
+As a result, we need to make APM server handle the compatibility with those for both existing and new agents.
+
+### Implementation phases
+
+Because there are a lots of moving pieces, implementation will be split into multiple phases:
+
+- **Phase 1** : APM server ingest (8.3)
+  - Spans intake: `span.context.service.target.*`, store them as-is in ES Span documents
+  - Transactions intake: add `service_target_type` and `service_target_name` next to `destination_service_resource` in `transaction.dropped_spans_stats` array, the related metrics documents should include `span.service.target.type` and `span.service.target.name` fields.
+  - TODO : is there anything required to ingest metrics with extra dimensions ?
```

Comment:
For the server side span metrics aggregation no metrics related changes are required on the Intake side, but the aggregation needs to be changed.

Reply:
When agents are sending to existing APM server, would it be an issue if those extra fields are provided in the span metrics ? If yes, that would make backwards compatibility with APM server easier.

Reply:
After implementing it in the Java agent I can confirm that sending those extra fields in the dropped spans metrics has no effect on current APM server, so we can mark this point as resolved.